src/main/java/com/selfxdsd/selfweb/api/ProjectsApi.java

```diff
@@ -189,4 +193,49 @@ public ProjectsApi(final User user, final Self self) {
         return ResponseEntity.ok(contracts.toString());
     }
 
+    /**
+     * Add new contributor for project Contract in Self.<br><br>
+     * @param owner Owner of the project (username or org name).
+     * @param name Simple name of the project.
+     * @param input Contract form input.
+     * @return Created Contract as JSON.
+     */
+    @PostMapping(
+        value = "/projects/{owner}/{name}/contracts",
+        produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public ResponseEntity<String> contracts(
+        @PathVariable("owner") final String owner,
+        @PathVariable("name") final String name,
+        @Valid final ContractInput input){
+
+        final String repoFullName = owner + "/" + name;
+        final String provider = this.user.provider().name();
+        final BigDecimal hourlyRate = BigDecimal
+            .valueOf(input.getHourlyRate())
+            .multiply(BigDecimal.valueOf(100));
+
+        ResponseEntity<String> response;
+        try {
+            final Contract contract = this.user
+                .projects()
+                .getProjectById(repoFullName, provider)
+                .contracts()
+                .addContract(repoFullName, input.getUsername(),
+                    provider, hourlyRate, input.getRole());
+            response = ResponseEntity
+                .status(HttpStatus.CREATED)
+                .body(new JsonContract(contract).toString());
+        } catch (final IllegalStateException exception) {
+            response = ResponseEntity
+                .status(HttpStatus.PRECONDITION_FAILED)
+                .body(Json.createObjectBuilder()
+                    .add("reason", exception.getMessage())
```

Comment:
@criske we should log the exception and exception message, but not return it in the response. It may contain some sensitive information :) 

In the response message you can just say "Something went wrong.".